### ISSUE TITLE
add pro controller support

### ISSUE THREAD

**Issue by JJRoyale** [redphx/joydance#43]
as bad as it sounds, my joy cons are broken and i play jd with the nx pro controller, can you add this to joydance?

**Comment by JJRoyale** [comment]
@redphx

**Comment by BroJac5246** [comment]
This sounds like a really good way to break your Pro Controller. Even if this were to be added (which I doubt), I would advise against this.

Also, how did you break both of your Joy Cons?

**Comment by JJRoyale** [comment]
i dont care, im pretty careful + if you think like that then your phone
could break too by playing

Il giorno dom 11 giu 2023 alle ore 16:45 BroJac5246 <
***@***.***> ha scritto:

**Comment by JJRoyale** [comment]
> Also, how did you break both of your Joy Cons?

I dunno, all of my joy cons either don't charge, have stick drift, they just don't work for gaming.

**Comment by BroJac5246** [comment]
Why did you reopen this?

**Comment by superpawko** [comment]
I would also like to see pro controller support. Why? Because Just Dance Watch from aliexpress pretend to be a Pro controller. I was able to use it with Yuzu and  BetterJoy.

**Comment by BroJac5246** [comment]
> I would also like to see pro controller support. Why? Because Just Dance Watch from aliexpress pretend to be a Pro controller. I was able to use it with Yuzu and  BetterJoy.  

Sure, but it's also an unofficial product— it's not representative of the actual experience with a Pro Controller. And there's no reason to add support just for some random Chinese product.

**Comment by superpawko** [comment]
Yes i know. Sorry. I am in a dreamland. But I dont have a possibility to update joycons. And I find it expensive to buy whole console just to play 5 hours a week. 

I wish your software can use any motion controller, like BetteryJoy creates motion server  for cemu. 

Could you just add functionality to use keyboard and use motion controllers on 127.0.0.1:26760?

Probably not. Thanks anyway!

**Comment by BroJac5246** [comment]
> Yes i know. Sorry. I am in a dreamland. But I dont have a possibility to update joycons. And I find it expensive to buy whole console just to play 5 hours a week. 

Do you have any friends who would let you update your Joy-Cons using their Switch?

> I wish your software can use any motion controller, like BetteryJoy creates motion server  for cemu. 
> 
> Could you just add functionality to use keyboard and use motion controllers on 127.0.0.1:26760?
> 
> Probably not. Thanks anyway! 

I would create separate issues for those, after checking for duplicates. They're more likely to be seen that way. However, redphx has made it pretty clear that JoyDance is only for Joy-Cons and anyone who wants to extend that functionality can fork the project. There's not much either of us can do about that short of forking the repo, since this project is barely maintained anyways.

**Comment by justasierof** [comment]
Just easy, in pycon folder, edit constants.py and change the joycon vendor id, joycon l product id, joycon r product id with the pro controller vendor and product id

**Comment by BroJac5246** [comment]
> Just easy, in pycon folder, edit constants.py and change the joycon vendor id, joycon l product id, joycon r product id with the pro controller vendor and product id

Did that work for you?

**Comment by justasierof** [comment]
Yes, but I only tried with jd2017pc , i don't know if the buttons will work

**Comment by BroJac5246** [comment]
> Yes, but I only tried with jd2017pc , i don't know if the buttons will work

The right ones might, though the left ones probably won't. Still, that's interesting.
